readthedocs/projects/tasks/search.py

```diff
@@ -40,112 +42,215 @@ def fileify(version_pk, commit, build, search_ranking, search_ignore):
         version_slug=version.slug,
     )
     try:
-        _create_imported_files(
+        _create_imported_files_and_search_index(
             version=version,
-            commit=commit,
-            build=build,
+            build_id=build,
             search_ranking=search_ranking,
             search_ignore=search_ignore,
         )
     except Exception:
         log.exception('Failed during ImportedFile creation')
 
+
+@app.task(queue="reindex")
+def index_build(build_id):
+    """Create imported files and search index for the build."""
+    build = (
+        Build.objects.filter(pk=build_id)
+        .select_related("version", "version__project")
+        .first()
+    )
+    if not build:
+        return
```

Comment:
I'd like to have `log.debug` at least on these so we can understand why indexes are not being created.

```suggestion
    if not build:
        log.debug("Skipping search indexing. Build object doesn't exists.")
        return
```

There are more `return`s we should add the logs.